features/apt_messages.feature

```diff
@@ -669,6 +669,348 @@ Feature: APT Messages
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    And I run `pro refresh messages` with sudo
+    And I apt upgrade
+    Then stdout matches regexp:
+      """
+      #
+      # one
+      # two
+      #
+      """
+    When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+      """
+      {
+        "messages": [
+          {
+            "begin": "$behave_var{today}",
+            "selectors": {
+              "codenames": ["mantic"],
+              "architectures": ["amd64"],
+              "packages": [["libcurl4", "==", "8.2.1-1ubuntu3"]]
+            },
+            "lines": [
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    And I run `pro refresh messages` with sudo
+    And I apt upgrade
+    Then stdout matches regexp:
+      """
+      #
+      # one
+      # two
+      #
+      """
+    # Testing multiple selectors together
+    When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+      """
+      {
+        "messages": [
+          {
+            "begin": "$behave_var{today}",
+            "selectors": {
+              "codenames": ["jammy"],
+              "architectures": ["arm64"],
+              "packages": [["libcurl4", "==", "8.2.1-1ubuntu3"]]
+            },
+            "lines": [
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    And I run `pro refresh messages` with sudo
+    And I apt upgrade
+    Then I will see the following on stdout:
+      """
+      Reading package lists...
+      Building dependency tree...
+      Reading state information...
+      Calculating upgrade...
+      The following packages have been kept back:
+        libcurl4
+      0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
+      """
+    When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+      """
+      {
+        "messages": [
+          {
+            "begin": "$behave_var{today}",
+            "selectors": {
+              "codenames": ["mantic"],
+              "architectures": ["amd64"],
+              "packages": [["libcurl4", ">", "8.2.1-1ubuntu3"]]
+            },
+            "lines": [
+              "one",
+              "two"
+            ]
+          }
+        ]
+      }
+      """
+    And I run `pro refresh messages` with sudo
+    And I apt upgrade
+    Then I will see the following on stdout:
+      """
+      Reading package lists...
+      Building dependency tree...
+      Reading state information...
+      Calculating upgrade...
+      The following packages have been kept back:
+        libcurl4
+      0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
+      """
+
+    Examples: ubuntu release
+      | release | machine_type  |
+      | mantic  | lxd-container |
```

Comment:
I think we should test this on all LTS releases as well

Reply:
Updated the test to support all LTS releases @lucasmoura